doc/making-a-visualizer.md

```diff
@@ -1,17 +1,194 @@
+validation in [`inhabit()`](#advanced-inhabit), as described below, but you
+won't be able to prompt the user for corrections at that point.
+
+-   **p5 Visualizer Template:** Make sure that the step size is positive.
+-   **Differences:** Make sure that the number of terms is at least the number
+    of levels.
+
+#### :nut*and_bolt: Inhabit a page element *(advanced)\_
+
+Each time the visualizer is inserted into a page element, the `inhabit()`
+function is called, giving you access to the element the visualizer is about
+to inhabit. If you implement `inhabit()`, start by passing the given element
+up to `super.inhabit()`, which does the behind-the-scenes work of managing p5
+canvass.
+
+When `inhabit()` is called, the visualizer will always be attached to a
+sequence, so you can do sequence-dependent validation and initialization here.
+
+-   **p5 Visualizer Template:** Go to the beginning of the sequence.
+-   **Differences:** Fill in the default value of the `levels` property in
+    case it was left unset in the parameters dialog. Do a consistency check
+    between the sequence being visualized and the parameters.
+
+#### :bulb: Set up the visualizer _(often used)_
+
+When the p5 graphics context becomes available, `setup()` is called, giving
+you your first chance to set graphics options and draw on the canvas. This is
+a good place for one-time graphics operations, like painting the background of
+a static visualizer or setting colors and stroke options that won't change
+from frame to frame.
+
+If you implement `setup()`, start by calling `super.setup()`, which includes
+the [`createCanvas()`](https://p5js.org/reference/#/p5/createCanvas) call that
+must appear in every p5 setup function.
+
+#### :nut*and_bolt: Show or stop the visualization; depart from a page element *(advanced)\_
+
+You shouldn't need to implement `show()`, `stop()`, or `depart()`. You can
+learn about them from the [visualizer interface](#abstract-visualizers)
+documentation, and from how they're implemented in the `P5Visualizer` base
+class.
+
+#### :key: Draw the visualization _(required)_
+
+The `draw()` function is called on each frame, giving you a chance to draw
+your visualization! Look at the examples and the p5
+[tutorials](https://p5js.org/learn/) and
+[p5.js reference](https://p5js.org/reference/) to learn about what you can do.
+
+Drawing tools and options are found in the graphics context, `this.sketch`.
+Accessing `this.sketch` triggers some consistency checks, so it's most
+parsimonious to access it at most once per function call, storing its value as
+a local constant for future reference.
+
+Sequence information is found in `this.seq`. This is a `SequenceInterface`
+object, so it will always have a method `getElement(n)` that returns the `n`th
+element in the sequence.
+
+You have to implement this method, even if it does nothing. Your visualizer
+can't be loaded into Numberscope without it.
```

Comment:
say "the `draw()` method" instead of "this method" since you just talked about the method `getElement`.